cloudconfig/templates.go

```diff
@@ -122,25 +94,23 @@ function downloadAndExtractRunner() {
 	# chown {{ .RunnerUsername }}:{{ .RunnerGroup }} -R /home/{{ .RunnerUsername }}/actions-runner/ || fail "failed to change owner"
 }
 
-CACHED_RUNNER=$(getCachedToolsPath)
-if [ -z "$CACHED_RUNNER" ];then
+RUN_HOME="/home/{{ .RunnerUsername }}/actions-runner"
+if [! -d "$RUN_HOME" ];then
 	downloadAndExtractRunner
 	sendStatus "installing dependencies"
-	cd /home/{{ .RunnerUsername }}/actions-runner
+	cd "$RUN_HOME"
 	sudo ./bin/installdependencies.sh || fail "failed to install dependencies"
 else
-	sendStatus "using cached runner found in $CACHED_RUNNER"
+	sendStatus "using cached runner found in $RUN_HOME"
 	OFS_AVAIL=1
-	RUN_HOME="/home/{{ .RunnerUsername }}/actions-runner"
 	sudo mkdir -p $OFS_DIR/upper-layer $OFS_DIR/work-layer $RUN_HOME
-	sudo chown {{ .RunnerUsername }}:{{ .RunnerGroup }} -R $OFS_DIR/upper-layer $OFS_DIR/work-layer $CACHED_RUNNER $RUN_HOME
-	sudo mount -t overlay overlay -o lowerdir=$CACHED_RUNNER,upperdir=$OFS_DIR/upper-layer,workdir=$OFS_DIR/work-layer $RUN_HOME || OFS_AVAIL=0
+	sudo chown {{ .RunnerUsername }}:{{ .RunnerGroup }} -R $OFS_DIR/upper-layer $OFS_DIR/work-layer $RUN_HOME $RUN_HOME
```

Comment:
I think we can skip the chown. If it's cached, it should already have proper ownership. Also, the OFS_DIR thing can be removed. There is no upper-layer or work-layer if we don't use overlay.

Reply:
You can also remove the mount -t overlay things. Just check if the folder is there, if not, download and install.

Reply:
Removed the part regarding the check for `.ExtraContext.OFS_DIR ` as well, since the only place it was used, was the cached runner